todo_app/app.py

```diff
@@ -80,12 +85,15 @@ def _callback():
         user = get_user(user_id)
         login_user(user)
         
+        logger.info("Logged user %s" % user_id)
+
         return redirect('/', code=302)
 
     @app.route('/useradmin', methods=['GET'])
     @login_required
     def _useradmin():
         if not is_permitted('admin'):
+            logger.warning("Unauthorised User %s attempted to access /useradmin" % current_user.id)
```

Comment:
Logging a warning when there is an auth issue is definitely a good idea - this line will only get hit if there is a bug giving a non admin a link to this site or someone unauthorised is trying their luck.

But for the same reason, you should also warn for all the other endpoints in the case where a reader is attempting a write action.